beacon_node/beacon_chain/src/historical_blocks.rs

```diff
@@ -62,10 +60,7 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
         &self,
         mut blocks: Vec<AvailableBlock<T::EthSpec>>,
     ) -> Result<usize, Error> {
-        let anchor_info = self
-            .store
-            .get_anchor_info()
-            .ok_or(HistoricalBlockError::NoAnchorInfo)?;
+        let anchor_info = self.store.get_anchor_info();
```

Comment:
The case where backfill is already complete is handled by the early return below for an empty batch.

I could add a sanity check error here instead if desired like:

```rust
if anchor.oldest_block_slot == 0 {
   return Err(HistoricalBlockError::AlreadyComplete);
}
```

?

Reply:
Don't see why we need the additional condition. Sync should be able to handle backfill already being complete